File: app/src/main/java/com/example/gaijin/countriestocities/dataclasses/GeonamesInfo.java

package com.example.gaijin.countriestocities.dataclasses;

import java.util.List;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

public class GeonamesInfo {

    @SerializedName("totalResultsCount")
    @Expose
    private Integer totalResultsCount;
    @SerializedName("geonames")
    @Expose
    private List<Geoname> geonames = null;

    public Integer getTotalResultsCount() {
        return totalResultsCount;
    }

    public void setTotalResultsCount(Integer totalResultsCount) {
        this.totalResultsCount = totalResultsCount;
    }

    public List<Geoname> getGeonames() {
        return geonames;
    }

    public void setGeonames(List<Geoname> geonames) {
        this.geonames = geonames;
    }

    @Override
    public String toString() {
        String info = "";
        for (Geoname geo : geonames) {
            info += geo.toString();
        }
        return info;
    }

}

File: app/src/main/java/com/example/gaijin/countriestocities/dataclasses/GeonamesInfo.kt
package com.example.gaijin.countriestocities.dataclasses
import com.google.gson.annotations.SerializedName


data class GeonamesInfo(
    @SerializedName("geonames")
    var geonames: List<Geoname>
)

data class Geoname(
    @SerializedName("countryCode")
    var countryCode: String,
    @SerializedName("elevation")
    var elevation: Int,
    @SerializedName("feature")
    var feature: String,
    @SerializedName("geoNameId")
    var geoNameId: Int,
    @SerializedName("lang")
    var lang: String,
    @SerializedName("lat")
    var lat: Double,
    @SerializedName("lng")
    var lng: Double,
    @SerializedName("rank")
    var rank: Int,
    @SerializedName("summary")
    var summary: String,
    @SerializedName("thumbnailImg")
    var thumbnailImg: String,
    @SerializedName("title")
    var title: String,
    @SerializedName("wikipediaUrl")
    var wikipediaUrl: String
)